### ISSUE TITLE
Status on CS1720 warning for extension methods

### ISSUE THREAD

**Issue by FlorianRappl** [dotnet/roslyn#1845]
In the past there has been a connect bug on displaying CS1720 warnings for extension methods. The original report can be viewed on https://connect.microsoft.com/VisualStudio/feedback/details/695362/c-compiler-gives-incorrect-warning-expression-will-always-cause-a-system-nullreferenceexception-for-extension-methods.

Basically the issued bug can be described by the following question on StackOverflow: http://stackoverflow.com/questions/26663240/cs1720-warning-erroneously-generated-for-extension-method. The discussion also shows the major problem: two identical approaches, where one yields a warning, the other does not.

The latest Visual Studio CTP also ships with a C# compiler that does not differentiate between a regular method and an extension method.

![image](https://cloud.githubusercontent.com/assets/1766191/7033763/f7a5cb10-dd7c-11e4-87e0-91c3efdd4748.png)

So long story short: Is it planned to fix this behavior? Previously it was not fixed, however, I feel that the warning is just wrong ("will always cause a NRE") and should (and could) be fixed.

Thanks!

**Comment by AlekseyTs** [comment]
@VSadov, please verify.
